docs/notebooks/batch_optimization.pct.py

```diff
@@ -0,0 +1,185 @@
+
+# %%
+from trieste.utils.objectives import branin, mk_observer, BRANIN_MINIMUM
+from trieste.acquisition.rule import OBJECTIVE
+from trieste.space import Box
+
+observer = mk_observer(branin, OBJECTIVE)
+search_space = Box([0, 0], [1, 1])
+
+num_initial_points = 5
+initial_query_points = search_space.sample(num_initial_points)
+initial_data = observer(initial_query_points)
+
+# %% [markdown]
+# ## Surrogate model
+# Just like in purely sequential optimization, we fit a surrogate Gaussian process model to the initial data.
+
+# %%
+import gpflow
+from trieste.models import create_model
+from trieste.utils import map_values
+
+
+def build_model(data):
+    variance = tf.math.reduce_variance(data.observations)
+    kernel = gpflow.kernels.Matern52(variance=variance, lengthscales=[0.2, 0.2])
+    gpr = gpflow.models.GPR(data.astuple(), kernel, noise_variance=1e-5)
+    gpflow.set_trainable(gpr.likelihood, False)
+
+    return {
+        OBJECTIVE: {
+            "model": gpr,
+            "optimizer": gpflow.optimizers.Scipy(),
+            "optimizer_args": {
+                "minimize_args": {"options": dict(maxiter=100)},
+            },
+        }
+    }
+
+
+model_spec = build_model(initial_data[OBJECTIVE])
+models = map_values(create_model, model_spec)
+
+# %% [markdown]
+# ## Batch acquisition functions.
+# To perform batch BO, we must define a batch acquisition function. Two popular batch acquisition functions supported in Trieste are `BatchMonteCarloExpectedImprovement` and `LocallyPenalizedExpectedImprovement`. Although both of these acquisition functions recommend batches of diverse query points, the batches are chosen in very different ways. `BatchMonteCarloExpectedImprovement` jointly allocates the batch of points as those with the largest expected improvement over our current best solution. In contrast, `LocallyPenalizedExpectedImprovement` greedily builds the batch, sequentially adding the maximizers of the standard (non-batch) `ExpectedImprovement` function penalized around the current pending batch points. In practice, `BatchMonteCarloExpectedImprovement` can be expected to have superior performance for small batches (`batch_size`<10) but scales poorly for larger batches.
+#
+# Note that both of these acquisition functions have controllable parameters. In particular, `BatchMonteCarloExpectedImprovement` is computed using a Monte-Carlo method (so it requires a `sample_size`), but uses a reparametrisation trick to make it deterministic. `LocallyPenalizedExpectedImprovement` has parameters controlling the degree of penalization that must be estimated from a random sample of `num_samples` model predictions.
+#
+# We now visualize the batch of 10 points chosen by each of these methods overlayed on the standard `ExpectedImprovement` acquisition function. `BatchMonteCarloExpectedImprovement` chooses a more diverse set of points, whereas `LocallyPenalizedExpectedImprovement` focuses evaluations in the most promising areas of the space.
```

Comment:
maybe shuffle the blocks a bit, to have `chosen_points_qei` and `chosen_points_lp` next to each other, and the text starting with "We now..." just before everything that's used for plotting? (including the `ei_acq_function` bit)